client/src/components/TestAuth/index.js

```diff
@@ -0,0 +1,32 @@
+import React, { useContext } from 'react';
+import { context } from '../userProvider';
+
+function TestAuth() {
+  const [userData, setUserData] = useContext(context);
+  const handleLogin = async () => {
+    const req = await fetch('https://jsonplaceholder.typicode.com/posts/1');
+    await req.json();
+    setUserData({ ...userData, isAuthenticated: true });
+  };
+  const handleLogout = async () => {
+    const req = await fetch('https://jsonplaceholder.typicode.com/posts/1');
+    await req.json();
+    setUserData({ ...userData, isAuthenticated: false });
```

Comment:
It might be better to empty out all user data when logging out, to prevent any possible errors where someone has logged out but there are components on-screen which are still displaying their name, for example.
`setUserData({isAuthenticated: false});`

Reply:
Ok, it's really important to rest all of the user properties , I'll change it;
Thank You Tom

Reply:
👍 You're welcome